google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataClient.java

```diff
@@ -159,6 +162,129 @@ public static BigtableDataClient create(BigtableDataSettings settings) throws IO
+  }
+
+  /**
+   * Confirms synchronously if given row key exists or not.
+   *
+   * <p>Sample code:
+   *
+   * <pre>{@code
+   * try (BigtableDataClient bigtableDataClient = BigtableDataClient.create("[PROJECT]", "[INSTANCE]")) {
+   *   String tableId = "[TABLE]";
+   *   ByteString key = ByteString.copyFromUtf8("key");
+   *
+   *   boolean isRowPresent = bigtableDataClient.exists(tableId, key);
+   *
+   *   // Do something with result, for example, display a message
+   *   if(isRowPresent) {
+   *     System.out.println(key.toStringUtf8() + " is present");
+   *   }
+   * } catch(ApiException e) {
+   *   e.printStackTrace();
+   * }
+   * }</pre>
+   *
+   * @throws com.google.api.gax.rpc.ApiException when a serverside error occurs
+   */
+  public boolean exists(String tableId, ByteString rowKey) {
+    return ApiExceptions.callAndTranslateApiException(existsAsync(tableId, rowKey));
+  }
+
+  /**
+   * Confirms asynchronously if given row key exists or not.
+   *
+   * <p>Sample code:
+   *
+   * <pre>{@code
+   * try (BigtableDataClient bigtableDataClient = BigtableDataClient.create("[PROJECT]", "[INSTANCE]")) {
+   *   String tableId = "[TABLE]";
+   *   final String key = "key";
+   *
+   *   ApiFuture<Boolean> futureResult = bigtableDataClient.existsAsync(tableId, key);
+   *
+   *   ApiFutures.addCallback(futureResult, new ApiFutureCallback<Boolean>() {
+   *     public void onFailure(Throwable t) {
+   *       t.printStackTrace();
+   *     }
+   *     public void onSuccess(Boolean isRowPresent) {
+   *       if(isRowPresent) {
+   *         System.out.println(key + " is present");
+   *       }
+   *     }
+   *   }, MoreExecutors.directExecutor());
+   * }
+   * }</pre>
+   */
+  public ApiFuture<Boolean> existsAsync(String tableId, String rowKey) {
+    return existsAsync(tableId, ByteString.copyFromUtf8(rowKey));
+  }
+
+  /**
+   * Confirms asynchronously if given row key exists or not.
+   *
+   * <p>Sample code:
+   *
+   * <pre>{@code
+   * try (BigtableDataClient bigtableDataClient = BigtableDataClient.create("[PROJECT]", "[INSTANCE]")) {
+   *   String tableId = "[TABLE]";
+   *   final ByteString key = ByteString.copyFromUtf8("key");
+   *
+   *   ApiFuture<Boolean> futureResult = bigtableDataClient.existsAsync(tableId, key);
+   *
+   *   ApiFutures.addCallback(futureResult, new ApiFutureCallback<Boolean>() {
+   *     public void onFailure(Throwable t) {
+   *       t.printStackTrace();
+   *     }
+   *     public void onSuccess(Boolean isRowPresent) {
+   *       if(isRowPresent) {
+   *         System.out.println(key.toStringUtf8() + " is present");
+   *       }
+   *     }
+   *   }, MoreExecutors.directExecutor());
+   * }
+   * }</pre>
+   */
+  public ApiFuture<Boolean> existsAsync(String tableId, ByteString rowKey) {
+    Query query =
+        Query.create(tableId).rowKey(rowKey).filter(Filters.FILTERS.limit().cellsPerRow(1));
```

Comment:
please chain values().strip() as well

Reply:
Thanks for the review!! Have updated with same PTAL